homeassistant/components/recorder/statistics.py

```diff
@@ -767,27 +802,26 @@ def get_metadata(
         )
 
 
-@overload
-def _configured_unit(unit: None, units: UnitSystem) -> None:
-    ...
-
+def _volume_unit(units: UnitSystem) -> str:
+    """Return the preferred volume unit according to unit system."""
+    if units.is_metric:
+        return VOLUME_CUBIC_METERS
+    return VOLUME_CUBIC_FEET
 
-@overload
-def _configured_unit(unit: str, units: UnitSystem) -> str:
-    ...
 
+def _configured_unit(
+    unit: str | None, state_unit: str | None, units: UnitSystem
+) -> str | None:
+    """Return the pressure and temperature units configured by the user.
 
-def _configured_unit(unit: str | None, units: UnitSystem) -> str | None:
-    """Return the pressure and temperature units configured by the user."""
-    if unit == PRESSURE_PA:
-        return units.pressure_unit
-    if unit == TEMP_CELSIUS:
-        return units.temperature_unit
+    Energy and volume is normalized for the energy dashboard.
+    For other units, display in the unit of the source.
+    """
+    if unit == ENERGY_KILO_WATT_HOUR:
```

Comment:
I wonder for energy dashboard if we could allow an option (in future PR) to override the unit to use, so the energy dashboard works, but otherwise (Lovelace cards etc) it's the original unit.

Reply:
That's planned for a follow-up!